functions/src/chat.ts

```diff
@@ -19,44 +20,50 @@ export async function deleteLobbyChatMessages(lobbyDoc: DocumentReference<Lobby>
   await batch.commit();
 }
 
-export async function deleteChatRooms(lobbyData: Lobby, lobbyDoc: DocumentReference<Lobby>, transaction: Transaction) {
+export async function deleteChatCollections(lobbyDoc: firestore.DocumentReference<Lobby>) {
+  const batch = db.batch();
+
+  const chatRooms = await getChatRoomCollection(lobbyDoc).get();
+
+  for (const chatRoomDoc of chatRooms.docs) {
+    // place messages in batch
+    const chatMessages = await getChatRoomMessagesCollection(chatRoomDoc.ref).get();
+    chatMessages.docs.map((messageDoc) => batch.delete(messageDoc.ref));
```

Comment:
this does not return anything so it should be a map.

maybe conside a for of loop